### ISSUE TITLE
[zsh-completion] CJK characters are not decoded when completed

### ISSUE THREAD

**Issue by whenov** [junegunn/fzf#551]
- Category
    - [x] fzf binary
    - [ ] fzf-tmux script
    - [ ] Key bindings
    - [x] Completion
    - [ ] Vim
    - [ ] Neovim
    - [ ] Etc.
- OS
    - [ ] Linux
    - [x] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [ ] bash
    - [x] zsh
    - [ ] fish

When I use the completion in shell, the completed Chinese characters are not decoded, here is the screenshot:
<img width="185" alt="screen shot 2016-04-23 at 13 24 17" src="https://cloud.githubusercontent.com/assets/899899/14759468/1cace644-0957-11e6-8cee-92b123eb9c8a.png">

**Comment by junegunn** [comment]
Yeah, they are not correctly displayed, but they are equivalent to the original characters, right?

**Comment by whenov** [comment]
They are part of the encoding of the original characters:
<img width="332" alt="screen shot 2016-04-24 at 11 27 59" src="https://cloud.githubusercontent.com/assets/899899/14765328/abd04916-0a0f-11e6-82ef-4ef6a3969da6.png">

**Comment by whenov** [comment]
I'm a little surprised why you have never encountered this problem...
<img width="124" alt="screen shot 2016-04-24 at 11 33 30" src="https://cloud.githubusercontent.com/assets/899899/14765364/bc1a216a-0a10-11e6-946c-74fb800545ef.png">

**Comment by junegunn** [comment]
I'm aware of the issue. But although the characters are not displayed as expected, they do point to the original characters, so at least it doesn't break the workflow.

```sh
$ echo test > 테스트
$ cat **<TAB>
$ cat $'�\204\220�\205��\204\211�\205��\204\220�\205�'
test
```

**Comment by whenov** [comment]
My case:
``` bash
$ echo test > 테스트
$ cat **<TAB>
$ cat $'\341'
cat: ?: No such file or directory
```

**Comment by junegunn** [comment]
Hmm, found that it works on bash, but not on zsh.
`printf %q 테스트` on zsh seems to give the wrong output for unicode characters.

**Comment by whenov** [comment]
In my experiment, both bash and zsh can't handle CJK characters when executing `printf %q`.

I suggest to put the filename in single quotes after escaping the single quotes themselves in filename.

**Comment by junegunn** [comment]
Using `${(q)VAR}` instead of `printf %q` fixes the issue. I'll push a fix shortly.

**Comment by junegunn** [comment]
Fixed in 988c9bd9be48b487bccca2d1d90d67c341c3bbf3. Thanks.

**Comment by whenov** [comment]
You rock!
